drivers/sensor/bmi270/bmi270.h

```diff
@@ -105,6 +108,48 @@
 #define BMI270_INT_IO_CTRL_OUTPUT_EN	BIT(3) /* Output enabled */
 #define BMI270_INT_IO_CTRL_INPUT_EN	BIT(4) /* Input enabled */
 
+/* CRT Feature macro definitions */
+#define BMI270_G_TRIG_1_SELECT_POS    8
+#define BMI270_G_TRIG_1_SELECT_MASK   BIT(8)
+#define BMI270_G_TRIG_1_SELECT_CRT    0x01
+#define BMI270_G_TRIG_1_BLOCK_POS     9
+#define BMI270_G_TRIG_1_BLOCK_MASK    BIT(9)
+#define BMI270_G_TRIG_1_BLOCK_UNBLOCK 0x00
+
+#define BMI270_G_TRIG_1_BURST_LEN_POS   0
+#define BMI270_G_TRIG_1_BURST_LEN_MASK  0xFF
+#define BMI270_G_TRIG_1_BURST_LEN_VALUE 1
+
+#define BMI270_GEN_SET_1_NVM_PROG_PREP_POS  10
+#define BMI270_GEN_SET_1_NVM_PROG_PREP_MASK BIT(10)
+#define BMI270_GEN_SET_1_NVM_PROG_PREP_EN   0x01
+#define BMI270_NVM_PROG_EN                  0x02
+
+#define BMI270_GYR_GAIN_CONFIG_MASK        0x7FF
+#define BMI270_GYR_GAIN_CONFIG_UPD_MSK     BIT(11)
+#define BMI270_GYR_GAIN_CONFIG_UPD_POS     11
+#define BMI270_GYR_GAIN_CONFIG_UPD_ENABLE  1
+#define BMI270_GYR_USR_GAIN_MASK           0x7F
+#define BMI270_GYR_GAIN_EN_MSK             BIT(7)
+#define BMI270_GYR_GAIN_EN_POS             7
+#define BMI270_GYR_GAIN_EN                 0x01
+#define BMI270_GYR_GAIN_STATUS_SAT_X_POS   0
+#define BMI270_GYR_GAIN_STATUS_SAT_X_MASK  BIT(0)
+#define BMI270_GYR_GAIN_STATUS_SAT_Y_POS   1
+#define BMI270_GYR_GAIN_STATUS_SAT_Y_MASK  BIT(1)
+#define BMI270_GYR_GAIN_STATUS_SAT_Z_POS   2
+#define BMI270_GYR_GAIN_STATUS_SAT_Z_MASK  BIT(2)
+#define BMI270_GYR_GAIN_STATUS_G_TRIG_POS  3
+#define BMI270_GYR_GAIN_STATUS_G_TRIG_MASK (BIT(3) | BIT(4) | BIT(5))
+#define BMI270_FIFO_CONFIG_1_SENSORS_MSK   (BIT(6) | BIT(7))
+#define BMI270_GYR_CRT_CONF_RUNNING_MSK    0x04
+#define BMI270_GYR_CRT_CONF_RUNNING_POS    0x02
+#define BMI270_GYR_CRT_CONF_RUNNING_EN     0x01
+#define BMI270_GYR_CRT_CONF_RUNNING_DIS    0x00
+/* Sensor specific attributes */
+#define BMI270_SENSOR_ATTR_NVM_PROG        (SENSOR_ATTR_PRIV_START)
```

Comment:
these private attributes should be defined in an enumation within a separate file `zephyr/include/zephyr/drivers/sensor/bosch_bmi270.h`.

Reply:
hmmm, I'm not sure how those attribute defines would find their way to the application. without the separate bmi270.h file I receive an error since it is not visible to the application.

```
error: 'BMI270_SENSOR_ATTR_NVM_PROG' undeclared (first use in this function)
ret = sensor_attr_set(imu_internal_dev, SENSOR_CHAN_GYRO_XYZ, BMI270_SENSOR_ATTR_NVM_PROG,
```

SHT4x has an example of how this is implemented
https://github.com/zephyrproject-rtos/zephyr/blob/9bbf7e5a4806c158e0d5c19b816379ef02e0f3e3/include/zephyr/drivers/sensor/sht4x.h#L30

Reply:
Agree with @maxmclau , the sm351lt and ina3221 drivers are wrong and need to be fixed